src/foremast/datapipeline/datapipeline.py

```diff
@@ -0,0 +1,96 @@
+
+import logging
+
+import awscli.customizations.datapipeline.translator as translator
+import boto3
+from tryagain import retries
+
+from ..utils import get_details, get_properties
+
+LOG = logging.getLogger(__name__)
+
+
+class AWSDataPipeline(object):
+    """Manipulate Data Pipeline."""
+
+    def __init__(self, app=None, env=None, region='us-east-1', prop_path=None):
+        """AWS Data Pipeline object.
+
+        Args:
+            app (str): Application name
+            env (str): Environment/Account
+            region (str): AWS Region
+            prop_path (str): Path of environment property file
+        """
+        self.app_name = app
+        self.env = env
+        self.region = region
+        self.properties = get_properties(prop_path)
+        self.datapipeline_data = self.properties[self.env]['datapipeline']
+        generated = get_details(app=self.app_name)
+        self.group = generated.data['project']
+
+        session = boto3.Session(profile_name=self.env, region_name=self.region)
+        self.client = session.client('datapipeline')
+        self.pipeline_id = None
+
+    def create_datapipeline(self):
+        """Creates the data pipeline if it does not already exist"""
+
+        tags = [
+                {"key": "app_group",
+                "value": self.group},
+                {"key": "app_name",
+                "value": self.app_name}
+                ]
+        response = self.client.create_pipeline(name=self.datapipeline_data.get('name', self.app_name),
+                                               uniqueId=self.app_name,
+                                               description=self.datapipeline_data['description'],
+                                               tags=tags)
+        self.pipeline_id = response.get('pipelineId')
+        LOG.debug(response)
+        LOG.info("Successfully configured Data Pipeline - %s", self.app_name)
+
+    def set_pipeline_definition(self):
+        """Translates the json definition and puts it on created pipeline"""
+
+        if not self.pipeline_id:
+            self.get_pipeline_id()
+
+        pipelineObjects = translator.definition_to_api_objects(self.datapipeline_data['json_definition'])
+        parameterObjects = translator.definition_to_api_parameters(self.datapipeline_data['json_definition'])
+        parameterValues = translator.definition_to_parameter_values(self.datapipeline_data['json_definition'])
+        response = self.client.put_pipeline_definition(
+                                pipelineId=self.pipeline_id,
+                                pipelineObjects=pipelineObjects,
+                                parameterObjects=parameterObjects,
+                                parameterValues=parameterValues)
+        LOG.debug(response)
+        LOG.info("Successfully applied pipeline definition")
+
+    def get_pipeline_id(self):
+        """Finds the pipeline ID for configured pipeline"""
+
+        response = self.client.list_pipelines()
```

Comment:
you're not guaranteed to get all of the results on the first call:  http://boto3.readthedocs.io/en/latest/reference/services/datapipeline.html#DataPipeline.Client.list_pipelines